mail_post_defer/models/mail_thread.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2022-2024 Moduon Team S.L. <info@moduon.team>
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+from datetime import datetime, timedelta
+
+from odoo import fields, models
+from odoo.exceptions import UserError
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    def _notify_thread(self, message, msg_vals=False, **kwargs):
+        """Defer emails by default."""
+        # Don't defer automatically if forcing send
+        _self = self
+        if "mail_defer_seconds" not in _self.env.context:
+            force_send = _self.env.context.get("mail_notify_force_send") or kwargs.get(
+                "force_send", False
+            )
+            kwargs.setdefault("force_send", force_send)
+            if not force_send:
+                # If deferring message, give the user some minimal time to revert it
+                _self = _self.with_context(mail_defer_seconds=30)
+        # Apply deferring
+        defer_seconds = _self.env.context.get("mail_defer_seconds")
+        if defer_seconds:
+            kwargs.setdefault(
+                "scheduled_date",
+                datetime.utcnow() + timedelta(seconds=defer_seconds),
```

Comment:
To be safe, since `datetime.utcnow()` is deprecated in Python 3.12.
```suggestion
                fields.datetime.now() + timedelta(seconds=defer_seconds),
```

Reply:
It make sense, I added `fields.Datetime.now()`